integration/qbft/tests/scenario_test.go

```diff
@@ -0,0 +1,194 @@
+
+		//initiating validators
+		for id := 1; id <= s.Committee; id++ {
+			id := spectypes.OperatorID(id)
+			s.validators[id] = createValidator(t, ctx, id, getKeySet(s.Committee), s.shared.Logger, s.shared.Nodes[id])
+		}
+
+		//invoking duties
+		for id, dutyProp := range s.Duties {
+			go func(id spectypes.OperatorID, dutyProp DutyProperties) { //launching goroutine for every validator
+				time.Sleep(dutyProp.Delay)
+
+				duty := createDuty(getKeySet(s.Committee).ValidatorPK.Serialize(), dutyProp.Slot, dutyProp.ValidatorIndex, role)
+				require.NoError(t, s.validators[id].StartDuty(duty))
+			}(id, dutyProp)
+		}
+
+		//validating state of validator after invoking duties
+		for id, validationFunc := range s.ValidationFunctions {
+			identifier := spectypes.NewMsgID(getKeySet(s.Committee).ValidatorPK.Serialize(), role)
+			//getting stored state of validator
+			var storedInstance *protocolstorage.StoredInstance
+			for {
+				var err error
+				storedInstance, err = s.validators[id].Storage.Get(spectypes.MessageIDFromBytes(identifier[:]).GetRoleType()).GetHighestInstance(identifier[:])
+				require.NoError(t, err)
+
+				if storedInstance != nil {
+					break
+				}
+
+				time.Sleep(500 * time.Millisecond) // waiting for duty will be done and storedInstance would be saved
+			}
+
+			// logging stored instance
+			jsonInstance, err := json.Marshal(storedInstance)
+			require.NoError(t, err)
+			fmt.Println(string(jsonInstance))
+
+			//validating stored state of validator
+			validationFunc(t, s.Committee, storedInstance)
+		}
+
+		// teardown
+		for _, val := range s.validators {
+			require.NoError(t, val.Stop())
+		}
+	})
+}
+
+func getKeySet(committee int) *spectestingutils.TestKeySet {
+	switch committee {
+	case 1, 2, 3, 4, 5, 6:
+		return KeySet4Committee
+	case 7, 8, 9:
+		return KeySet7Committee
+	case 10, 11, 12:
+		return KeySet10Committee
+	case 13, 14, 15:
+		return KeySet13Committee
+	default:
+		panic("unsupported committee size")
+
+	}
+}
+
+func testingShare(keySet *spectestingutils.TestKeySet, id spectypes.OperatorID) *spectypes.Share { //TODO: check dead-locks
+	return &spectypes.Share{
+		OperatorID:      id,
+		ValidatorPubKey: keySet.ValidatorPK.Serialize(),
+		SharePubKey:     keySet.Shares[id].GetPublicKey().Serialize(),
+		DomainType:      spectypes.PrimusTestnet,
+		Quorum:          keySet.Threshold,
+		PartialQuorum:   keySet.PartialThreshold,
+		Committee:       keySet.Committee(),
+	}
+}
+
+func quorum(committee int) int {
+	return (committee*2 + 1) / 3 // committee = 3f+1; quorum = 2f+1 // https://drive.google.com/file/d/1bP_MLq0MM7ZBSR0Ddh7HUPcc42vVUKwz/view?usp=share_link
+}
+
+func newStores(logger *zap.Logger) *qbftstorage.QBFTStores {
+	db, err := storage.GetStorageFactory(basedb.Options{
+		Type:   "badger-memory",
+		Path:   "",
+		Logger: logger,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	store := qbftstorage.New(db, logger, "integration-tests", protocolforks.GenesisForkVersion)
```

Comment:
the prefix should be based on the role type  and not `"integration-tests"`
so need to have different instance for each role type

Reply:
resolved